src/main/scala/tricera/concurrency/CCReader.scala

```diff
@@ -623,12 +623,21 @@ class CCReader private (prog              : Program,
     }).toList
 
   // todo: only add types that exist in the program - should also add machine arithmetic types
-  val predefSignatures =
-    List(("O_Int", HeapObj.CtorSignature(List(("getInt", HeapObj.OtherSort(Sort.Integer))), ObjSort)),
-         ("O_UInt", HeapObj.CtorSignature(List(("getUInt", HeapObj.OtherSort(Sort.Nat))), ObjSort)),
-         ("O_Addr", HeapObj.CtorSignature(List(("getAddr", HeapObj.AddrSort)), ObjSort)),
-         ("O_AddrRange", HeapObj.CtorSignature(List(("getAddrRange", HeapObj.AddrRangeSort)), ObjSort))
-    )
```

Comment:
The following is an improvement, but still not ideal. Ideally, we collect which program types can be on the heap, and use only those types in `predefSignatures`.